deployments/tyk/docker-compose.yml

```diff
@@ -10,35 +10,38 @@ services:
       - ./deployments/tyk/volumes/tyk-dashboard/tyk_analytics.conf:/opt/tyk-dashboard/tyk_analytics.conf
       - ./deployments/tyk/volumes/tyk-dashboard/private-key.pem:/opt/tyk-dashboard/private-key.pem
     environment:
-      - TYK_DB_LICENSEKEY=${DASHBOARD_LICENCE}
-      - TYK_INSTRUMENTATION=${INSTRUMENTATION_ENABLED}
+      - TYK_DB_LICENSEKEY=${DASHBOARD_LICENCE:?Dashboard licence missing from Docker environment file .env. Review The Getting Started steps in README.md.}  
+      - TYK_LOGLEVEL=${DASHBOARD_LOGLEVEL:-info}
+      - TYK_INSTRUMENTATION=${INSTRUMENTATION_ENABLED:-0}
     depends_on:
       - tyk-redis
       - tyk-mongo
   tyk-gateway:
-    image: tykio/tyk-gateway:latest
+    image: tykio/tyk-gateway:${GATEWAY_VERSION:-latest}
     ports:
       - 8080:8080
     networks:
       - tyk
     environment:
-      - TYK_INSTRUMENTATION=${INSTRUMENTATION_ENABLED}
-      - TYK_GW_TRACER_ENABLED=${TRACING_ENABLED}
+      - TYK_LOGLEVEL=${GATEWAY_LOGLEVEL:-info}
+      - TYK_INSTRUMENTATION=${INSTRUMENTATION_ENABLED:-0}
+      - TYK_GW_TRACER_ENABLED=${TRACING_ENABLED:-0}
     volumes:
       - ./deployments/tyk/volumes/tyk-gateway/tyk.conf:/opt/tyk-gateway/tyk.conf
       - ./deployments/tyk/volumes/tyk-gateway/certs:/opt/tyk-gateway/certs
       - ./deployments/tyk/volumes/tyk-gateway/middleware:/opt/tyk-gateway/middleware
     depends_on:
       - tyk-redis
   tyk-gateway-2:
-    image: tykio/tyk-gateway:latest
+    image: tykio/tyk-gateway:${GATEWAY2_VERSION:-latest}
     ports:
       - 8081:8080
     networks:
       - tyk
     environment:
-      - TYK_INSTRUMENTATION=${INSTRUMENTATION_ENABLED}
-      - TYK_GW_TRACER_ENABLED=${TRACING_ENABLED}
+      - TYK_LOGLEVEL=${GATEWAY2_LOGLEVEL:-info}
```

Comment:
May be better to have a single log level env var, to switch debug on/off everywhere.

Reply:
sure

Reply:
@davegarvey I found out that actually it's not so convenient